spec/models/ldap_auth_source_spec.rb

```diff
@@ -252,45 +253,144 @@
         let(:attr_admin) { 'isAdmin' }
 
         it 'maps for the admin user in ldap', :aggregate_failures do
-          admin_attributes = ldap.find_user('ldap_admin')
-          expect(admin_attributes).to be_a Hash
-          expect(admin_attributes[:firstname]).to eq 'LDAP'
-          expect(admin_attributes[:lastname]).to eq 'Adminuser'
-          expect(admin_attributes[:admin]).to eq true
-
-          admin_attributes = ldap.find_user('bb459')
-          expect(admin_attributes).to be_a Hash
-          expect(admin_attributes[:firstname]).to eq 'Belle'
-          expect(admin_attributes[:lastname]).to eq 'Baldwin'
-          expect(admin_attributes[:admin]).to eq false
+          admin = ldap.find_user('ldap_admin')
+          expect(admin).to be_a User
+          expect(admin.firstname).to eq 'LDAP'
+          expect(admin.lastname).to eq 'Adminuser'
+          expect(admin.admin).to eq true
+
+          user = ldap.find_user('bb459')
+          expect(user).to be_a User
+          expect(user.firstname).to eq 'Belle'
+          expect(user.lastname).to eq 'Baldwin'
+          expect(user.admin).to eq false
+        end
+
+        context 'with an existing user and different attributes' do
+          let!(:user) { create(:user, ldap_auth_source: ldap, login: 'ldap_admin') }
+
+          it 'updates the user' do
+            expect(user.firstname).to eq 'Bob'
+            expect(user).not_to be_admin
+
+            user = ldap.find_user('ldap_admin')
```

Comment:
I would verify the persisted user instead.

```suggestion
            ldap.find_user('ldap_admin')
            user.reload
```